### PR TITLE
Redactor3

### PR DESCRIPTION
**Pull request by thet**
- Upgrade to Redactor 3
- Use Webpack as build environment.

This PR builds upon the redactor2 branch.

### CODE REVIEW

File: src/pat-redactor.js [hunk 1]
```diff
@@ -1,121 +1,191 @@
...
+      "properties",
+      "table",
+      "textdirection",
+      "textexpander",
+      "video"
+    ],
+    true
+  );
+  //parser.add_argument("allowed-tags", [], [], true);
+  //parser.add_argument("denied-tags", [], [], true);
```
Comment (thet) [inline src/pat-redactor.js]: I couldn't find a replacement for those two options

File: src/pat-redactor.js [hunk 2]
```diff
@@ -1,121 +1,191 @@
...
+  parser.add_argument("toolbar-type", "standard", ["standard", "fixed", "air"]);
+  parser.add_argument("toolbar-external", null);
+  parser.add_argument("toolbar-fixed-target", null);
+
+  parser.add_argument(
+    "buttons",
+    [
+      "html",
+      "undo",
+      "redo",
```
Comment (thet) [inline src/pat-redactor.js]: undo, redo and html are now normal buttons.
For html the ``show-source-button`` option needs to be checked.

File: src/pat-redactor.js [hunk 3]
```diff
@@ -1,121 +1,191 @@
...
+          "minHeight",
+          "formatting",
+          //"deniedTags",
+          //"allowedTags",
+        ])
+      );
 
-            // Until here
-            $el.redactor(this.initializePlugins(options)[0]);
+      //options.limiter = poptions.limitCharacters;
```
Comment (thet) [inline src/pat-redactor.js]: I thought this option was removed but it is a plugin now. Have to add this later...

File: src/pat-redactor.js [hunk 4]
```diff
@@ -1,121 +1,191 @@
...
+      "html",
+      "undo",
+      "redo",
+      "format",
+      "bold",
+      "italic",
+      "deleted",
+      "lists",
+      "link",
+      "line",
```
Comment (thet) [inline src/pat-redactor.js]: Was: ``horizontalrule`` in Redactor 2

File: src/pat-redactor.js [hunk 5]
```diff
@@ -1,121 +1,191 @@
...
+
+  parser.add_argument(
+    "formatting",
+    ["p", "blockquote", "pre", "h1", "h2", "h3", "h4", "h5"],
+    ["p", "blockquote", "pre", "h1", "h2", "h3", "h4", "h5"],
+    true
+  );
+
+  parser.add_argument(
+    "plugins",
```
Comment (thet) [inline src/pat-redactor.js]: No plugins for: bufferbuttions (undo and redo are normal buttons), source (now an option + button), codemirror (now an option for source option), romanlisting

**Comment by pilz** [comment]
Limiter is important for oira.
romanlisting was a plugin we wrote ourselves. Need to port that

File: src/pat-redactor.js (hunk 1 above)
Comment (pilz) [reply src/pat-redactor.js]: I believe this has been replaced by the clean function https://imperavi.com/redactor/docs/settings/clean/
I think we'll drop it here.

File: src/pat-redactor.js (hunk 2 above)
Comment (pilz) [reply src/pat-redactor.js]: Ok, changed in proto, works fine.

File: src/pat-redactor.js (hunk 4 above)
Comment (pilz) [reply src/pat-redactor.js]: Changed in proto, Cornelis needs to adapt styling.

File: src/pat-redactor.js (hunk 5 above)
Comment (pilz) [reply src/pat-redactor.js]: we must port our own romanlisting plugin before updating ikath!

**Pull request merged**